From 7f91509e793e429fac3c7bc5280ee5a522238b71 Mon Sep 17 00:00:00 2001
From: Jared Morrow <jm@joyent.com>
Date: Tue, 7 Aug 2018 18:13:17 +0000
Subject: [PATCH] MANTA-3774 Use fastdelete method to delete objects if user
 account has snaplinks disabled Reviewed by: Jan Wyszynski
 <jan.wyszynski@joyent.com> Approved by: Jan Wyszynski
 <jan.wyszynski@joyent.com>

---
 lib/obj.js | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/lib/obj.js b/lib/obj.js
index 4a5233a..7d82bb9 100644
--- a/lib/obj.js
+++ b/lib/obj.js
@@ -926,6 +926,22 @@ function deletePointer(req, res, next) {
     };
 
     log.debug(opts, 'deletePointer: entered');
+
+    /*
+     * Let the delete mechanism know that snaplinks are disabled for this
+     * account, so it can treat the object differently
+     */
+    var uuid = req.owner.account.uuid;
+    common.checkAccountSnaplinksEnabled(req, uuid, function (enabled) {
+        // Pass in a special header if they are not enabled
+        if (!enabled) {
+            log.debug({
+                link: req.link,
+                owner: req.owner.account
+            }, 'deletePointer: owner of object has snaplinks disabled');
+            opts.snapLinksDisabled = true;
+        }
+    });
     req.moray.delMetadata(opts, function (err) {
         if (err) {
             next(err);
-- 
2.21.0

